### PR TITLE
Adopt "Keep a Changelog" format

### PR DESCRIPTION
**Pull request by ratijas**
Looks kinda reasonable. Will it work?

Used by https://github.com/rust-lang/rust/blob/master/src/bootstrap/CHANGELOG.md and (although not explicitly stated) by sixtyfps.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+## [Unreleased]
+
+### Added
+- Tutorial on adding Rust wrappers #171.
+
+### Deprecated
+- Rename QObjectDescription in favor of its new name RustQObjectDescriptor #172.
+
+### Removed
+- Support for QT_SELECT environment variable when running qmake #173.
```
Comment (ogoffart) [inline CHANGELOG.md:16]:
```suggestion
- No longer set QT_SELECT environment variable when running qmake #173.
```

**Review by ogoffart — APPROVED** [review]

**Comment by ratijas** [comment]
On a project management note, is it OK to squash and merge single-commit PRs? I feel like branches are a good way to keep track of individual steps of ongoing/finished work, but single commit merges just double the number of commits without noticeable benefits — especially when branched off the tip of master.

**Comment by ogoffart** [comment]
Yes, you can rebase,  linear history is nicer.

**Comment by ratijas** [comment]
Unlike "rebase" button though, "squash and merge" also adds this cool hashtag `(#179)` by default. Which, I guess, is better for GitHub integration.

**Pull request merged**